ConfigSpace/hyperparameters.pyx

```diff
@@ -1538,88 +1555,129 @@ cdef class UniformIntegerHyperparameter(IntegerHyperparameter):
             return False
 
     def get_num_neighbors(self, value = None) -> int:
-        return self.upper - self.lower
+        # If there is a value in the range, then that value is not a neighbor of itself
+        # so we need to remove one
+        if value is not None and self.lower <= value <= self.upper:
+            return self.upper - self.lower - 1
+        else:
+            return self.upper - self.lower
```

Comment:
I wasn't sure how to handle this but I thought it should act similar to `Categorical` and that the `value` would count as a neighbor. However Categorical just implies it implicitly, regardless of the `value` passed in. Here, I've said that any `value` in the range can not be a neighbor of itself.